### PR TITLE
Add 'settingtypes.txt'

### PR DESCRIPTION
**Pull request by AntumDeluge**
Add 'settingtypes.txt' file for configuring mod from client GUI.

### CODE REVIEW

File: settingtypes.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+[mesecons_blinkyplant]
+
+mesecon.blinky_plant_interval (Plant interval) int 3
+
+
+[mesecons_detector]
+
+mesecon.detector_radius (Detector radius) int 6
+mesecon.node_detector_distance_max (Detector max distance) int 10
```
Comment (numberZero) [inline settingtypes.txt]: “Player detector” and “Node detector”, I suppose

File: settingtypes.txt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+
+[mesecons_luacontroller]
+
+mesecon.luacontroller_string_rep_max (Max) int 64000
+mesecon.luacontroller_digiline_maxlen (Digiline max length) int 50000
+mesecon.luacontroller_maxevents (Max events) int 10000
+mesecon.luacontroller_memsize (Memory size) int 100000
```
Comment (numberZero) [inline settingtypes.txt]: - `string:rep` result length limit
- Digiline message size limit
- Events per ...?
- Add units (bytes (?))

File: settingtypes.txt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+[mesecons_pistons]
+
+mesecon.piston_max_push (Max push) int 15
+mesecon.piston_max_pull (Max pull) int 15
+
+
+[mesecons_pressureplates]
+
+mesecon.pplate_interval (Interval) float 0.1
```
Comment (numberZero) [inline settingtypes.txt]: “Check interval”?

**Comment by AntumDeluge** [comment]
Added some of the recommended changes by @numberZero. But "Max events" is still the same. Not sure what to put there. numberZero recommended something like "Events per ?".

**Comment by numberZero** [comment]
@AntumDeluge er, only some my recommendations were literal descriptions. You should have read them, instead of copy-pasting blindly.

**Comment by AntumDeluge** [comment]
I knew they were suggestions, I just didn't know what else to put there. Was waiting for more suggestions.

**Comment by numberZero** [comment]
@AntumDeluge As you see, I changes some long names. Look at the settings list (in MT settings menu, not in editor, to make experiment cleaner) and ask yourself whether you understand what each option does just by reading its name. If so, names are probably good; otherwise, improvement may be needed.

P.S. Parentheses are illegal in setting names)

**Comment by SmallJoker** [comment]
This PR is now open for a quite long time - how mergeable is it right now? All setting descriptions except `mesecon.resumetime` are clear to me. Some kind of delay in seconds?

File: settingtypes.txt [hunk 4]
```diff
@@ -0,0 +1,43 @@
+[mesecons]
+
+mesecon.resumetime (Resume time) int 4
```
Comment (SmallJoker) [inline settingtypes.txt]: Could be called `Startup delay time`. Minimal value 1s, max should be somewhere around 10s
The settings don't seem to be limited in the code, but let's at least define some limits here.

**Comment by numberZero** [comment]
@SmallJoker is this OK now?

**Comment by AntumDeluge** [comment]
Thank you for the help everyone. 😃

**Pull request merged**